torch/ao/quantization/fx/_lower_to_native_backend.py

```diff
@@ -130,9 +137,149 @@ def _lower_weighted_ref_module(model: QuantizedGraphModule) -> QuantizedGraphMod
         model.recompile()
     return model
 
+def _lower_quantized_binary_op(
+    model: QuantizedGraphModule,
+    qconfig_map: Dict[str, QConfigAny]
+) -> QuantizedGraphModule:
+    modules = dict(model.named_modules(remove_duplicate=False))
+
+    def get_bop_patterns(bop: Callable) -> Pattern:
+        return [
+            (torch.quantize_per_tensor,
+             (bop, "dequantize", "dequantize"),
+             MatchAllNode, MatchAllNode, MatchAllNode)
+            for bop_pattern in [
+                (bop, "dequantize", "dequantize"),
+                (bop, MatchAllNode, "dequantize"),
+                (bop, "dequantize", MatchAllNode)
+            ]
+        ]
+
+    def get_bop_relu_patterns(bop: Callable) -> List[Pattern]:
+        return [
+            (torch.quantize_per_tensor,
+             (relu_op, bop_pattern),
+             MatchAllNode, MatchAllNode, MatchAllNode)
+            for relu_op in [torch.relu, torch.nn.functional.relu, torch.nn.ReLU]
+            for bop_pattern in [
+                (bop, "dequantize", "dequantize"),
+                (bop, MatchAllNode, "dequantize"),
+                (bop, "dequantize", MatchAllNode),
```

Comment:
Do we need (bop, "dequantize", "dequantize"), since anything that matches this one would also match these two patterns?

Reply:
god point, this is referring to add/add_scalar, maybe we could just use MatchAllNode for both arguments, and filter the things we want later, this would reduce the number of patterns.